internal/scheduler/legacyscheduler.go

```diff
@@ -765,7 +824,12 @@ func pickQueueRandomly(weights map[string]float64, random *rand.Rand) (string, f
 	for _, share := range weights {
 		sum += share
 	}
-	pick := sum * random.Float64()
+	var pick float64
+	if random != nil {
```

Comment:
why do we support passing a nil random into this function?

Reply:
For this function, I think it's reasonable to fall back to the default if a rand isn't provided.